### ISSUE TITLE
Use Android real device in debug mode, Surface::setbuffersDimensions Log being spammed 

### ISSUE THREAD

**Issue by cixiu** [flutter/flutter#28502]
## Steps to Reproduce
1. flutter create flutter_test_demo
2. connect andriod real device
3. flutter run
4. click FloatingActionButton
5. watching terminal, terminal are spammed by `Surface::setbuffersDimensions` Log 

![issue1](https://user-images.githubusercontent.com/26104985/53402562-eb0be080-39ec-11e9-8f49-f563cc7da7a2.png)

## Logs
1. flutter doctor -v
![image](https://user-images.githubusercontent.com/26104985/53402831-6b324600-39ed-11e9-874c-2bce70971b2a.png)

**Comment by zoechi** [comment]
I wasn't able to reproduce. Can you please try with Flutter from `dev` channel?

```
flutter channel dev
flutter doctor
```

I tried with 
```
[✓] Flutter (Channel master, v1.2.3-pre.47, on Mac OS X 10.14.3 18D42, locale en-AT)
[✓] Connected device (2 available)
    • Pixel XL                  • HT69V0203649  • android-arm64 • Android 9 (API 28)
    • Android SDK built for x86 • emulator-5554 • android-x86   • Android 9 (API 28) (emulator)
```

**Comment by cixiu** [comment]
@zoechi It seems that this problem only appears in the lower version of Android.
When I switched Android 5.0 to Android 7.0 or more higher Android version, the problem did not recur.
```
[√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Version 10.0.17134.590], locale zh-CN)
[√] Connected device (1 available)                                                                                                   
    • m1 note • 71UBBL622SZC • android-arm64 • Android 5.1 (API 22)
```

**Comment by TahaTesser** [comment]
Issue isn't reproduced on

<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, v1.12.13+hotfix.8, on Microsoft Windows [Version 10.0.19041.84], locale en-US)
    • Flutter version 1.12.13+hotfix.8 at C:\Code\flutter_stable
    • Framework revision 0b8abb4724 (4 weeks ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\Code\SDK
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = C:\Code\SDK
    • Java binary at: C:\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.6)
    • Android Studio at C:\Code\android-studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] VS Code (version 1.43.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (1 available)
    • SM M305F • 32003c30dc19668f • android-arm64 • Android 10 (API 29)

• No issues found!
```

</details>

<details>
 <summary>Code Sample</summary>

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.display1,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```


</details>

<details>
 <summary>Logs</summary>

```
Launching lib\main.dart on SM M305F in debug mode...
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        42.0s
√ Built build\app\outputs\apk\debug\app-debug.apk.
Installing build\app\outputs\apk\app.apk...                         6.5s
D/FlutterActivity( 4881): Using the launch theme as normal theme.
D/FlutterActivityAndFragmentDelegate( 4881): Setting up FlutterEngine.
D/FlutterActivityAndFragmentDelegate( 4881): No preferred FlutterEngine was provided. Creating a new FlutterEngine for this FlutterFragment.
D/FlutterActivityAndFragmentDelegate( 4881): Attaching FlutterEngine to the Activity that owns this Fragment.
D/FlutterView( 4881): Detaching from a FlutterEngine: null
D/FlutterView( 4881): Not attached to an engine. Doing nothing.
D/FlutterView( 4881): Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@9ac55ca
D/FlutterActivityAndFragmentDelegate( 4881): Executing Dart entrypoint: main, and sending initial route: /
Syncing files to device SM M305F...                                     
 6,936ms (!)

🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
An Observatory debugger and profiler on SM M305F is available at: http://127.0.0.1:53706/SKrH4IVGwF4=/
For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
D/ViewRootImpl@66ce36c[MainActivity]( 4881): stopped(false) old=true
D/SurfaceView( 4881): windowStopped(false) true io.flutter.embedding.android.FlutterSurfaceView{5e5d817 V.E...... ........ 0,0-1080,2301} of ViewRootImpl@66ce36c[MainActivity]
D/SurfaceView( 4881): onWindowVisibilityChanged(0) true io.flutter.embedding.android.FlutterSurfaceView{5e5d817 V.E...... ........ 0,0-1080,2301} of ViewRootImpl@66ce36c[MainActivity]
W/libEGL  ( 4881): EGLNativeWindowType 0x7c132caa50 disconnect failed
D/ViewRootImpl@66ce36c[MainActivity]( 4881): Relayout returned: old=(0,0,1080,2340) new=(0,0,1080,2340) req=(1080,2340)0 dur=21 res=0x40001 s={true 532634222592} ch=false
D/ViewRootImpl@66ce36c[MainActivity]( 4881): stopped(false) old=false
D/FlutterView( 4881): Attaching to a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@9ac55ca
D/FlutterView( 4881): Already attached to this engine. Doing nothing.
D/ViewRootImpl@66ce36c[MainActivity]( 4881): Relayout returned: old=(0,0,1080,2340) new=(0,0,1080,2340) req=(1080,2340)0 dur=16 res=0x40001 s={true 532634222592} ch=false
D/OpenGLRenderer( 4881): createReliableSurface : 0x7c132caa40, 0x7c03794000
D/mali_winsys( 4881): EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
D/ViewRootImpl@66ce36c[MainActivity]( 4881): MSG_WINDOW_FOCUS_CHANGED 1 1
D/InputMethodManager( 4881): prepareNavigationBarInfo() DecorView@2ddd340[MainActivity]
D/InputMethodManager( 4881): getNavigationBarColor() -855310
D/InputMethodManager( 4881): prepareNavigationBarInfo() DecorView@2ddd340[MainActivity]
D/InputMethodManager( 4881): getNavigationBarColor() -855310
V/InputMethodManager( 4881): Starting input: tba=com.example.stable_demo ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
D/InputMethodManager( 4881): startInputInner - Id : 0
I/InputMethodManager( 4881): startInputInner - mService.startInputOrWindowGainedFocus
D/InputMethodManager( 4881): prepareNavigationBarInfo() DecorView@2ddd340[MainActivity]
D/InputMethodManager( 4881): getNavigationBarColor() -855310
V/InputMethodManager( 4881): Starting input: tba=com.example.stable_demo ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
D/InputMethodManager( 4881): startInputInner - Id : 0
D/InputMethodManager( 4881): prepareNavigationBarInfo() DecorView@2ddd340[MainActivity]
D/InputMethodManager( 4881): getNavigationBarColor() -855310
V/InputMethodManager( 4881): Starting input: tba=com.example.stable_demo ic=null mNaviBarColor -855310 mIsGetNaviBarColorSuccess true , NavVisible : true , NavTrans : false
D/InputMethodManager( 4881): startInputInner - Id : 0
I/InputMethodManager( 4881): startInputInner - mService.startInputOrWindowGainedFocus
D/InputTransport( 4881): Input channel destroyed: 'ClientS', fd=88
D/ViewRootImpl@66ce36c[MainActivity]( 4881): stopped(true) old=false
D/SurfaceView( 4881): windowStopped(true) false io.flutter.embedding.android.FlutterSurfaceView{5e5d817 V.E...... ........ 0,0-1080,2301} of ViewRootImpl@66ce36c[MainActivity]
D/SurfaceView( 4881): surfaceDestroyed callback.size 1 #1 io.flutter.embedding.android.FlutterSurfaceView{5e5d817 V.E...... ........ 0,0-1080,2301}
W/libEGL  ( 4881): EGLNativeWindowType 0x7b80146010 disconnect failed
D/SurfaceView( 4881): remove() io.flutter.embedding.android.FlutterSurfaceView{5e5d817 V.E...... ........ 0,0-1080,2301} Surface(name=SurfaceView - com.example.stable_demo/com.example.stable_demo.MainActivity@5e5d817@0)/@0xafd106e
D/FlutterView( 4881): Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@9ac55ca
E/ViewRootImpl@66ce36c[MainActivity]( 4881): Surface is not valid.
D/ViewRootImpl@66ce36c[MainActivity]( 4881): MSG_WINDOW_FOCUS_CHANGED 0 1
D/InputMethodManager( 4881): prepareNavigationBarInfo() DecorView@2ddd340[MainActivity]
D/InputMethodManager( 4881): getNavigationBarColor() -855310

```

</details>



if you're still experiencing this problem, please create a new issue and provide the information required in the template.
Thank you
